servers/zts/src/main/java/com/yahoo/athenz/zts/ZTSImpl.java

```diff
@@ -3526,8 +3526,19 @@ void validateInstanceServiceIdentity(DomainData domainData, final String service
         // need to check anything
 
         final String domainName = domainData.getName();
-        if (validateServiceSkipDomains.contains(domainName)) {
-            return;
+        if (domainName != null) {
+            for (String serviceSkipDomain : validateServiceSkipDomains) {
+                // first, we perform validation using wildcards
+                if (serviceSkipDomain.endsWith("*")) {
+                    String serviceSkipDomainPrefix = serviceSkipDomain.substring(0, serviceSkipDomain.length() - 1);
+                    if (domainName.startsWith(serviceSkipDomainPrefix)) {
+                        return;
+                    }
+                } else if (validateServiceSkipDomains.contains(domainName)) {
```

Comment:
wrong check here to verify against the whole list every time. The correct line would be 

} else if (skipDomain.equals(domainName)) {

Reply:
Thank you. I've fixed it.